In [2]:
!pip3 install OpenNMT-py

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 51.8/51.8 kB 3.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.1/62.1 kB 4.3 MB/s eta 0:00:00
INFO: pip is looking at multiple versions of mkl-fft to determine which version is compatible with other requirements. This could take a while.
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.0/62.0 kB 5.1 MB/s eta 0:00:00
INFO: pip is looking at multiple versions of mkl-random to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of mkl-umath to determine which version is compatible with other requirements. This could take a while.
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 262.8/262.8 kB 11.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.6/38.6 MB 40.6 MB/s eta 0:00:00:00:0100:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.0/17.0 MB 24.7 MB/s eta 0:00:0000:0100:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 755.6/755.6 MB 628.7

In [3]:
!pip install "numpy<2"

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.0/61.0 kB 2.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.3/18.3 MB 54.2 MB/s eta 0:00:00:00:010:01m
  Attempting uninstall: numpy
    Found existing installation: numpy 2.2.6
    Uninstalling numpy-2.2.6:
      Successfully uninstalled numpy-2.2.6
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
bigframes 2.8.0 requires google-cloud-bigquery-storage<3.0.0,>=2.30.0, which is not installed.
gensim 4.3.3 requires scipy<1.14.0,>=1.7.0, but you have scipy 1.15.3 which is incompatible.
datasets 3.6.0 requires fsspec[http]<=2025.3.0,>=2023.1.0, but you have fsspec 2025.5.1 which is incompatible.
onnx 1.18.0 requires protobuf>=4.25.1, but you have protobuf 3.20.3 which is incompatible.
cesium 0.12.4 requires numpy<3.0,>=2.0, but you have numpy 1.26.4 which is incompatible.
google-colab 1.0.0 requires 

In [ ]:
from huggingface_hub import hf_hub_download

model_path = hf_hub_download(
    repo_id="LULab/whisper-align-chn-checkpoints-alignment",
    filename="checkpoints/transformer.mmspell_step_200000.pt"
)

print(f"Model downloaded to: {model_path}")


checkpoints/transformer.mmspell_step_300(…):   0%|          | 0.00/577M [00:00<?, ?B/s]

Model downloaded to: /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-align-chn-checkpoints-alignment-para-add-lora/snapshots/d1c0c76a055099144da8bb60b4e6ee41f9c0617f/checkpoints/transformer.mmspell_step_300000.pt


In [6]:
import pandas as pd
test_df = pd.read_csv("/kaggle/input/test-without-feat/Test_Without_Feat.tsv", sep="\t")
test_df

ASR_Error  \
0     အ သ စ လော့ တော ခြောင်း အ ဝါ သည် တွက် များ ကိုး...   
1     အ လောင်း သည် ရေ်း နေ ရာ တွင် ပေါ် လာ သည် မှာ သ...   
2     ဘူ ရှ် ၏ ပြော ၍ေး ဆို ခွ င့်း ရှိ သူ ဂျော ဒန် ...   
3     တိး ပ တ့် သူ မျော် တို့ အ ဗုဒ္တ ဘာ ဍာ သည် ဗုဒ္...   
4     များ စွာ သော ကိစ္စ ရက် များ တွင် နိုင် ငံ ရပ် ...   
...                                                 ...   
3180  ဗိုလ် ချုပ် ကြွန် စော ရဲ စာ အုပ် က ပြည် ဘက် မှ...   
3181  ထို့ နောက် က လေး နှစ် ဦး အား မည် ရ ထား လမ်း ပေ...   
3182  ကုမ္ပ ဏီ ဆိုင် ရာ သ တင်း အ ချက် အ လက် တွေ ကို ...   
3183    က လေး တွေ ဘောင်း မီး အ တို အ လေး တွေ ဝတ် ကြ တယ်   
3184  ထို စင် က စီး ပွား ရေး မ ပြေ လဲ လ သော သူ က ဝင်...   

                                                Correct  
0     အော့ ဆ လော့ တော ကြောင် အ ဝါ သည် ကြွက် များ ကို...  
1     အ လောင်း သည် ယင်း နေ ရာ တွင် ပေါ် လာ သည် မှာ တ...  
2     ဘု ရှ် ၏ ပြော ရေး ဆို ခွ င့် ရှိ သူ ဂျော ဒန် ဂ...  
3     တိ ဘက် လူ မျိုး တို့ ၏ ဗုဒ္ဓ ဘာ သာ သည် ဗုဒ္ဓ ၏...  
4     များ စွာ သော ကိစ္စ ရပ် များ တွင် နိုင် ငံ ရပ် ...  
...                                                 ...  
3180  ဗိုလ် ချုပ် ကျော် ဇော ရဲ့ စာ အုပ် က ပြည် ပ မှာ...  
3181  ထို့ နောက် က လေး နှစ် ဦး အား မီး ရ ထား လမ်း ဘေ...  
3182  ကုမ္မ ဏီ ဆိုင် ရာ သ တင်း အ ချက် အ လက် တွေ ကို ...  
3183       က လေး တွေ ဘောင်း ဘီ အ တို လေး တွေ ဝတ် ကြ တယ်  
3184  ထို စဉ် က စီး ပွား ရေး မ ပြေ လည် လှ သော သူ က ဝ...  

[3185 rows x 2 columns]

In [7]:
test_df["ASR_Error"].to_csv("err_syl1.test", index=False, header=False)

In [10]:
!onmt_translate \
  -model /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-align-chn-checkpoints-alignment-para-add-lora/snapshots/d1c0c76a055099144da8bb60b4e6ee41f9c0617f/checkpoints/transformer.mmspell_step_300000.pt \
  -src /kaggle/working/err_syl1.test \
  --beam_size 20 \
  -output /kaggle/working/without_feat_align_v7_30.txt \
  -gpu 0


[2025-09-12 01:59:31,260 INFO] Loading checkpoint from /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-align-chn-checkpoints-alignment-para-add-lora/snapshots/d1c0c76a055099144da8bb60b4e6ee41f9c0617f/checkpoints/transformer.mmspell_step_300000.pt
[2025-09-12 01:59:32,120 INFO] Loading data into the model
[2025-09-12 02:01:48,395 INFO] PRED SCORE: -0.2962, PRED PPL: 1.34 NB SENTENCES: 3185
Time w/o python interpreter load/terminate:  137.14506149291992


[2025-09-06 13:09:19,823 INFO] PRED SCORE: -0.2867, PRED PPL: 1.33 NB SENTENCES: 3185
Time w/o python interpreter load/terminate:  139.26940870285034

[2025-09-06 03:10:54,283 INFO] PRED SCORE: -0.2873, PRED PPL: 1.33 NB SENTENCES: 3185
Time w/o python interpreter load/terminate:  140.4818470478058

In [20]:
!onmt_translate -model /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-synde-align-IPA-checkpoints/snapshots/3c8da246b8889591d9e56be2a1fec62fdfd2c6ed/checkpoints/transformer.mmspell_step_200000.pt \
    -src /kaggle/working/err_syl1.test \
    -n_src_feats 1 \
    --beam_size 20\
    -output /kaggle/working/align_synde_ipa.txt \
    -gpu 0

[2025-08-24 15:36:09,045 INFO] Loading checkpoint from /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-synde-align-IPA-checkpoints/snapshots/3c8da246b8889591d9e56be2a1fec62fdfd2c6ed/checkpoints/transformer.mmspell_step_200000.pt
/usr/local/lib/python3.11/dist-packages/onmt/modules/embeddings.py:240: UserWarning: Not merging with sum and positive feat_vec_size, but got non-default feat_vec_exponent. It will be unused.
  warnings.warn(
/usr/local/lib/python3.11/dist-packages/onmt/modules/embeddings.py:240: UserWarning: Not merging with sum and positive feat_vec_size, but got non-default feat_vec_exponent. It will be unused.
  warnings.warn(
[2025-08-24 15:36:10,016 INFO] Loading data into the model
[2025-08-24 15:38:18,047 INFO] PRED SCORE: -0.3231, PRED PPL: 1.38 NB SENTENCES: 3185
Time w/o python interpreter load/terminate:  129.01350569725037


In [8]:
!onmt_translate \
    -model /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-G2P-checkpoints/snapshots/7bfc05e5316b993eb11cce6d5f974f087d18d1d5/checkpoints/transformer.mmspell_step_200000.pt \
    -src /kaggle/working/err_syl1.test \
    -n_src_feats 1 \
    -beam_size 20 \
    -output /kaggle/working/g2p_chn.txt \
    -gpu 0

[2025-08-23 10:16:17,401 INFO] Loading checkpoint from /root/.cache/huggingface/hub/models--YeBhoneLin10--whisper-G2P-checkpoints/snapshots/7bfc05e5316b993eb11cce6d5f974f087d18d1d5/checkpoints/transformer.mmspell_step_200000.pt
/usr/local/lib/python3.11/dist-packages/onmt/modules/embeddings.py:240: UserWarning: Not merging with sum and positive feat_vec_size, but got non-default feat_vec_exponent. It will be unused.
  warnings.warn(
/usr/local/lib/python3.11/dist-packages/onmt/modules/embeddings.py:240: UserWarning: Not merging with sum and positive feat_vec_size, but got non-default feat_vec_exponent. It will be unused.
  warnings.warn(
[2025-08-23 10:16:19,889 INFO] Loading data into the model
[2025-08-23 10:18:40,272 INFO] PRED SCORE: -0.2891, PRED PPL: 1.34 NB SENTENCES: 3185
Time w/o python interpreter load/terminate:  142.886488199234
